drivers/audio/dmic_nrfx_pdm.c

```diff
@@ -18,14 +18,23 @@ LOG_MODULE_REGISTER(dmic_nrfx_pdm, CONFIG_AUDIO_DMIC_LOG_LEVEL);
 #if CONFIG_SOC_SERIES_NRF54HX
 #define DMIC_NRFX_CLOCK_FREQ MHZ(16)
 #define DMIC_NRFX_CLOCK_FACTOR 8192
+#define DMIC_NRFX_AUDIO_CLOCK_FREQ DT_PROP_OR(DT_NODELABEL(audiopll), frequency, 0)
 #else
 #define DMIC_NRFX_CLOCK_FREQ MHZ(32)
 #define DMIC_NRFX_CLOCK_FACTOR 4096
+#define DMIC_NRFX_AUDIO_CLOCK_FREQ \
+	COND_CODE_1(CONFIG_SOC_NRF54L20, \
+		(DT_PROP_OR(DT_NODELABEL(aclk), clock_frequency, 0)), \
+		(DT_PROP_OR(DT_NODELABEL(clock), hfclkaudio_frequency, 0)))
 #endif
 
 struct dmic_nrfx_pdm_drv_data {
 	const nrfx_pdm_t *pdm;
+#if CONFIG_CLOCK_CONTROL_NRF
 	struct onoff_manager *clk_mgr;
+#elif CONFIG_CLOCK_CONTROL_NRF2_AUDIOPLL
+	const struct device *audiopll_dev;
```

Comment:
it would be better to have this in config structure as it can be initialized at compile time (contrary to `clk_mgr`).